node/service_setup.go

```diff
@@ -45,7 +45,7 @@ func (node *Node) setupForExplorerNode() {
 	)
 	// Register explorer service.
 	node.serviceManager.RegisterService(
-		service.SupportExplorer, explorer.New(&node.SelfPeer),
+		service.SupportExplorer, explorer.New(&node.SelfPeer, node.IsInSync),
```

Comment:
this doesn't work well because IsInSync will only be updated every 60seconds

```
func (node *Node) DoSyncing(bc *core.BlockChain, worker *worker.Worker, willJoinConsensus bool) {
	ticker := time.NewTicker(time.Duration(SyncFrequency) * time.Second)
	// TODO ek – infinite loop; add shutdown/cleanup logic
	for {
		select {
		case <-ticker.C:
			node.doSync(bc, worker, willJoinConsensus)
		case <-node.Consensus.BlockNumLowChan:
			node.doSync(bc, worker, willJoinConsensus)
		}
	}
}```

The best approach is still do the checking freshly when the RPC is called. Otherwise, the RPC can return stale state of the sync (meaning it's not sync'ed, but IsInSync == true because it's stale)